alethzero/Transact.cpp

```diff
@@ -356,23 +446,19 @@ GasRequirements Transact::determineGasRequirements()
 			er = ethereum()->call(from, value(), to, m_data, mid, gasPrice(), PendingBlock, FudgeFactor::Lenient);
 		if (er.excepted == TransactionException::OutOfGas || er.excepted == TransactionException::OutOfGasBase || er.excepted == TransactionException::OutOfGasIntrinsic || er.codeDeposit == CodeDeposit::Failed)
 		{
-			lowerBound = mid;
-			if (!haveUpperBound)
-				upperBound *= 2;
+			m_startLowerBound = mid;
+			m_startUpperBound *= 2;
 		}
 		else
 		{
-			lastGood = er;
-			if (haveUpperBound)
-				upperBound = mid;
-			else
-				haveUpperBound = true;
+			m_lastGood = lastGood;
+			m_lowerBound = m_startLowerBound;
+			m_upperBound = m_startUpperBound;
+			startTimer(0);
```

Comment:
Can this be called while the previous timer is still active? Also startTimer in contructor uses different interval